cmd/puppetdb/pdb.go

```diff
@@ -90,30 +91,79 @@ func execute(api string, query string) {
 	cli.PrintString(data)
 }
 
-func parseInput(command string) (string, string) {
-	blocks := strings.Split(command, " ")
-	var api, query string
+func extractString(str, start, end string) (result string) {
+	s := strings.Index(str, start)
+	s--
+	if s == -1 {
+		return
+	}
+	s += len(start)
+	e := strings.LastIndex(str, end)
+	e++
+	if e == -1 {
+		return
+	}
+	return str[s:e]
+}
+
+func parseInput(command string) (string, string, puppetdb.Pagination) {
+
+	//Split Query Api and Pagination Up
+	checkForQuery, err := regexp.Match(`[\\w+]`, []byte(command))
+	var query string
+	if checkForQuery {
+		query = extractString(command, "[", "]")
+		if query == "[]" {
+			query = ""
+		}
+	}
+	if err != nil {
+		fmt.Println("No query parameters detected", err)
+	}
+
+	querylessCommand := strings.Replace(command, query, "", 1)
+
+	blocks := strings.Split(querylessCommand, " ")
+	api := blocks[0]
 
-	switch blocks[0] {
-	case "exit":
+	if api == "exit" {
 		fmt.Println("Bye!")
 		os.Exit(0)
-	case "factnames":
-		api = blocks[0]
-	case "nodes", "facts", "inventory", "reports":
-		if len(blocks) < 2 {
-			fmt.Println("please provide a query or enter '[]' for none")
-			return "", ""
+	}
+
+	var rex = regexp.MustCompile(`(\w+)=(\w+)`) //nolint
+
+	options := rex.FindAllString(querylessCommand, -1)
+
+	pagination := puppetdb.Pagination{}
+
+	for _, option := range options {
```

Comment:
I think it might be worth trying to extract from here to the bottom of the loop out into a separate function to help improve readability.